src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -454,15 +495,25 @@ def recordInCouch(self, jobs, newstate, oldstate, updatesummary=False):
                                 jobSummary[prop] = jobSummary[prop] if jobSummary[prop] else currentJobDoc.get(prop, [])
                         except CouchNotFoundError:
                             pass
+                    if countDocs >= self.fwjrdatabase.getQueueSize():
+                        self.jsumdatabase.commit()
                     self.jsumdatabase.queue(jobSummary, timestamp=True)
 
         if len(couchRecordsToUpdate) > 0:
             self.setCouchDAO.execute(bulkList=couchRecordsToUpdate,
                                      conn=self.getDBConn(),
                                      transaction=self.existingTransaction())
 
+        try:
+            self.fwjrdatabase.commit(callback=discardConflictingDocument)
+        except CouchRequestTooLargeError as exc:
+            msg = "Failed to commit bulk of framework job report to CouchDB."
+            msg += f" Error: {str(exc)}"
+            logging.warning(msg)
+            shrinkLargeFJR(self.fwjrdatabase, 8e6)
+            # now all the documents should fit in
+            self.fwjrdatabase.commit(callback=discardConflictingDocument)
```

Comment:
I just answered a similar question above. AFAIK, only the `fwjr` key can be as large as MB, hence this is the only key that we shrink in `shrinkLargeFJR`. If it is not, then we will get to know it in the future because docs will keep failing to be inserted in the database.

Having said that, at the moment I see no reason for a second check and if some exception happens, let it go all the way upstream.